[Solved #7] Write unit tests

supunTE opened the pull request:
### Solved #7 

Unit tests
- [x] Reducers
- [x] Effects
- [x] Utils
- [x] Loggers
- [x] Create docs

supunTE committed:
docs(sample state object): fix data models imports

supunTE committed:
build(nanoid): downgrade nanoid version to 3.3 from 4.0

downgrade nanoid to solve an error raise with jest

BREAKING CHANGE: package update

supunTE committed:
feat(data models): export sockets data models

supunTE committed:
test(reducers): create add operation reducer tests

re #7

supunTE committed:
test(reducers): spread up the check to equal whole game state

Spread up the checks to equal whole game state from store root level. It will check if any developer
add new store to the state.

re #7

supunTE committed:
refactor(initialstate): make initialState empty

Clear the initialState and make it empty. Moved populated data into sample state data object.

supunTE committed:
style: add new end line to app.module.css.map

supunTE committed:
build(jest): enable jest verbose

re #7

supunTE committed:
test(effects): write test for player join

re #7

supunTE committed:
fix(util): import TextEncoder into generate avatar link

Import TextEncoder into generate avatar link util to solve error showing in jest.

re #7

supunTE committed:
feat(store): create initialize store function to make mock stores

re #7

supunTE committed:
test(reducers): use fresh stores in test cases

re #7

supunTE committed:
Revert "test(reducers): spread up the check to equal whole game state"

This reverts commit c8fc1937a8b5d71a7e039f1f19968021e2f055f9.

supunTE committed:
test(reducers): spread up the check to equal whole game state

Spread up the checks to equal whole game state from store root level. Now the test will fail if a developer add new store to the game state.

re #7

supunTE committed:
build(package.json): add scripts to test all and test store

re #7

supunTE committed:
feat(reducers): add condition to check availability and existence of payload data

supunTE committed:
refactor(move leaderboardreducer from update to add): move leaderboardReducer from update to add

Change updateLeaderboardReducer to addLeaderboardReducer.

re #7

supunTE committed:
fix(reducer): provide correct payload interface to remove tournament reducer

Provide removeTournamentReducerPayload to removeTournamentReducer, which had
removePlayerReducerPayload mistakenly.

BREAKING CHANGE: Fix a bug in a reducer and it will change payload object of it.

re #5

supunTE committed:
fix(effects): change updateLeaderboardReducer to addLeaderboardReducer

re #5

supunTE committed:
refactor(client): remove auto generated test

re #7

supunTE committed:
build(nx cloud): initialize nx cloud

supunTE committed:
test(reducer): create add leaderboard test

re #7

supunTE committed:
test(reducers): create tests for remove operations in reducers

re #7

supunTE committed:
test(reducers): create tests for update operations in reducers

re #7

supunTE committed:
test(reducers): create tests for logger in reducer

re #7

supunTE committed:
build(jest): add jest option to generate codeCoverage

re #7

supunTE committed:
build(package.json): initialize lodash

supunTE committed:
feat(reducers): use lodash omit to remove element from object

Replace object desctructing method to lodash omit for removing element from object. This update will remove the unused const.

supunTE committed:
test(reducers): add more elements to the initial values of the remove tournament reducer test

re #7

supunTE committed:
test(reducers): add test for update first playerLog

re #7

supunTE committed:
feat(constants): initialize lib constants

re #7

supunTE committed:
chore(gitignore): update gitignore

supunTE committed:
fix(constants): initialize constant for max error logs count

supunTE committed:
feat(reducers): add more tests to the logger reducer

add test to check on reducer removing old logs properly

re #7

supunTE committed:
test(store): remove auto generated tests

re #7

supunTE committed:
test(effects): update tests for effects of player

Use fresh stores for test, tests to check whether logger raisers are working, tests to check for
joining existing tournament

re #7

supunTE committed:
refactor(reducers): remove console logs

supunTE committed:
test(reducers): add initial state locally

re #7

supunTE committed:
test(reducers): write tests to cover complete code

Write tests to complete test code coverage including minor branches on reducers

re #7

supunTE committed:
feat(loggers): add logger raisers for missing payloads

re #7

supunTE committed:
feat(effects): implement more logger raisers on effects

re #7

supunTE committed:
test(effects): add tests for clear player

re #7

supunTE committed:
refactor(raisers): change loggers to raisers

re #7

supunTE committed:
test(mocks): create mock data of players

create mock data of players to use in test cases.

re #7

supunTE committed:
test(mocks): add more mock data

Add more mock data for players, tournament, race. Create functions to generate mock data.

re #7

supunTE committed:
test(effects): replace logger raisers directory

re #7

supunTE committed:
feat(effects): update data validation

Update max username length, add raiser to clearPlayer if id not found

re #7

supunTE committed:
test(effects): complete test code coverage of effects in player

re #7

supunTE committed:
build(package.json): install jest-when

Jest when package will allow to specify dynamic return values for specifically matched mocked
function arguments

re #7

supunTE committed:
test(mocks): update mocks

Add function to genrate player models, Make race mock data

re #7

supunTE committed:
refactor(effects): replace text api with sample text

Add sample race text because generating race text should move to server side later.

re #7

supunTE committed:
feat(effects): add method for changing tournament state to empty

Add method to clearPlayer effect to set tournament state to empty if no players available in the
tournament.

re #7

supunTE committed:
feat(effects): add more logger raisers and functionalties to race effects

re #7

supunTE committed:
test(effects): add more tests to player effects

Add test to check setting tournamnet state to empty if the last player of tournament left

re #7

supunTE committed:
test(test): add tests for effects of race

re #7

supunTE committed:
test(mocks): add more mock values

Add mock values for race texts, player logs

re #7

supunTE committed:
feat(payloads): update payloads

Add update player reducer payload, add race text to start countdown payload

re #7

supunTE committed:
feat(reducers): update reducers

add reducer to update player, remove unwanted console logs

re #7

supunTE committed:
test(reducers): add tests to update player effect

re #7

supunTE committed:
test(effects): add tests for logger effect

re #7

supunTE committed:
feat(effects): add effect to setState tournament

add effect to set state of the tournament and remove set ready effect

re #7

supunTE committed:
test(effects): add tests to effects

add tests for setTournamentState effect

re #7

supunTE committed:
feat(payloads): update setTournamentState payload

re #7

supunTE committed:
test(effects): add seperate initialState to the race tests

re #7

supunTE committed:
test(mocks): remove unwanted default values

re #7

supunTE committed:
test(raisers): add tests for logger raisers

re #7

supunTE committed:
build(package.json): remove unwanted packages

supunTE committed:
test(mock): update mocks

re #7

supunTE committed:
feat(constants): update constants

supunTE committed:
test(mocks): seperate mocks into a new library

re #7

supunTE committed:
test(mocks): seperate mocks into a new library

re #7

supunTE committed:
test(mocks): create a generate zero padding function inside mocks

re #7

supunTE committed:
test(mocks): create a generate zero padding function inside mocks

re #7

supunTE committed:
feat(data models): add more data models for leaderboard

Add specific data model for finished player and timeout player leaderboard entries

re #7

supunTE committed:
feat(data models): add more data models for leaderboard

Add specific data model for finished player and timeout player leaderboard entries

re #7

supunTE committed:
test(effects): change mocks import location

re #7

supunTE committed:
test(effects): change mocks import location

re #7

supunTE committed:
refactor(utils): refactor calculate timeout timer util code

Import constant, change local variable names.

supunTE committed:
refactor(utils): refactor calculate timeout timer util code

Import constant, change local variable names.

supunTE committed:
refactor(utils): change file name format  of extract id util

supunTE committed:
refactor(utils): change file name format  of extract id util

supunTE committed:
refactor(utils): refactor generate avatar link code

assign global text encoder, convert functions to arrow functions

supunTE committed:
refactor(utils): refactor generate avatar link code

assign global text encoder, convert functions to arrow functions

supunTE committed:
feat(utils): add sort functionality to generate leaderboard util

re #7

supunTE committed:
feat(utils): add sort functionality to generate leaderboard util

re #7

supunTE committed:
refactor(utils): change file name format of generate uid util

re #7

supunTE committed:
refactor(utils): change file name format of generate uid util

re #7

supunTE committed:
refactor(utils): update exporting utils as file names have been changed

re #7

supunTE committed:
refactor(utils): update exporting utils as file names have been changed

re #7

supunTE committed:
test(utils): write complete tests for utils

re #7

supunTE committed:
test(utils): write complete tests for utils

re #7

supunTE committed:
Merge branch 'feature/7-write-unit-tests-rebase' into feature/7-write-unit-tests

supunTE committed:
style: remove unwanted comments

supunTE committed:
style(eslint): move out common eslint rules to root dir

supunTE committed:
refactor(store and util): add return types for functions which missing return types

supunTE committed:
test(effects): fix failing test

re #7

supunTE committed:
feat(effects): add more conditions to joinPlayer effect

add one more condition to change tournament state to Lobby from Empty

re #7

supunTE committed:
test(effects): add tests to check converting tournament state to Lobby from Empty

re #7

supunTE committed:
refactor(raisers): remove unwanted return keywords

re #7

supunTE committed:
docs(reducers): add js docs for reducer functions

re #7

supunTE committed:
feat(husky): add command to run all test before pushing

supunTE committed:
test(reducers): give descriptive names for tests

re #7

supunTE committed:
docs(store): add jsdocs to store and initial state

re #7

supunTE committed:
docs(effects): add jsdocs

re #7

supunTE committed:
docs(reducers): fix typos and mistakes in js docs

re #7

supunTE committed:
test(reducers): remove unnecessary state checking

re #7

supunTE committed:
test(effects): give descriptive names for tests

re #7

supunTE committed:
fix(client): move store provider to inside of the strict mode

re #5

supunTE committed:
Merge branch 'feature/5-setup-state-manager' into feature/7-write-unit-tests

supunTE committed:
docs(data models): update js docs

supunTE committed:
Merge branch 'feature/5-setup-state-manager' into feature/7-write-unit-tests

supunTE committed:
refactor(effects): change effect setStateTournament to setTournamentState

re #5

supunTE committed:
fix(effects): remove explicit state types for payloads

re #5

supunTE committed:
fix(effects): remove unnecessary else statement for getting race index

re #5

supunTE committed:
fix(effects): remove unnecessary if statement

re #5

supunTE committed:
fix(store): remove unwanted async/await and add missing await keywords

re #5

supunTE committed:
feat(eslint): update eslint rules

supunTE committed:
feat(util): simplified util give zero padding

re #5

supunTE committed:
Merge branch 'feature/5-setup-state-manager' into feature/7-write-unit-tests

supunTE committed:
Merge branch 'feature/5-setup-state-manager' into feature/7-write-unit-tests

supunTE committed:
build(package.json): update npm scripts

add npm scripts to lint all, test store and util seperately

supunTE committed:
feat(util): add type map to define return type base on input parameter on extract ids

re #5

supunTE committed:
feat(eslint): add eslint rules to avoid unnecessary await

supunTE committed:
refactor(eslint): fix all eslint errorsRemove unnecessary await and async, add missing return types

supunTE committed:
docs(store): update jsdocsupdate docs after replacing some async function with normal functions

supunTE committed:
fix(reducers): update condition checks

remove unnecessary payload checks, add state availability checks where necessary

re #7

supunTE committed:
test(reducers): update tests

update tests for new checks in reducers

re #7

supunTE committed:
fix(effects): fix startCountdown effect using wrong reducer

Change startCountdown effect to use addRace reducer instead of updateRace reducer. Also the effect will
change tournament state using setTournamentState effect now.

re #7

supunTE committed:
refactor(store): remove unnecessary testing codes

supunTE committed:
refactor(raisers): simplify logic using logical operators

supunTE committed:
refactor(reducers/remove): simplify logic using filter method

supunTE committed:
refactor(util): change case of the enum extractIdType to pascal case

re #5

supunTE committed:
feat(util): add dynamic return type based on input parameter on generate uid function

re #5

supunTE committed:
refactor(effects/player): remove unnecessary explicit type

supunTE committed:
docs(effects): add jsdocs to start countdown effect

re #7

supunTE committed:
docs(payloads): add descriptions for payloads

re #7

supunTE committed:
refactor(utils): clean code of generate leaderboard

re #7

supunTE committed:
docs(util): created docs for utils

re #7

supunTE committed:
docs(payloads): update docs for payloads

supunTE committed:
docs(mocks): complete js docs for mocks

re #7

supunTE committed:
docs(raisers): complete jsdocs for logger raisers

re #7

supunTE committed:
refactor(store): refactor and clean codes

supunTE committed:
fix(effects/race): replace effect to reduce code

supunTE committed:
refactor(mocks): move mock error logs generator function to mocks from logger test file in reducers

re #7

supunTE committed:
fix(reducers/remove): fix removeTournament function looking for every current players log

removeTournament function was looking for every current player's player log which could be different
from relevant race players' player logs. The function will look for only players relevant to that
race now.

re #7

pasinducw commented on libs/constants/src/lib/constants.ts:
```diff
@@ -0,0 +1,11 @@
+export const MAX_ERR_LOGS_COUNT = 1024;
```
Add descriptions for each constant; together with some context to where it is being used.

supunTE replied:
Added descriptions 93015d626ca686a6707ee4f18bf61e385507952b

pasinducw commented on libs/util/src/lib/calculate-timeout-timer.ts:
```diff
@@ -0,0 +1,20 @@
+import { AVERAGE_WPM } from '@razor/constants';
+
+const averageWPM = AVERAGE_WPM;
```
Unnecessary variable

supunTE replied:
Removed d21e4c8dd553653a0e557a2563fa48eed875a216

pasinducw commented on libs/util/src/lib/calculate-timeout-timer.ts:
```diff
@@ -0,0 +1,20 @@
+/** Calculating timeout timer for the race text.
+ *
+ * @param {string} text - Race text.
+ * @returns {number} - Timeout timer in seconds.
```
What this function does is to compute the timeout *duration*, in seconds.
_timer_ is a tool that could probably take the duration as input and start ticking down.
https://www.google.com/search?q=timer&rlz=1C5CHFA_enLK1008LK1008&sxsrf=ALiCzsbWZjHkVKmg-8WgAHID-QgS0NdxWw:1670048344717&source=lnms&tbm=isch&sa=X&ved=2ahUKEwi1seDy5tz7AhVsR2wGHQFUDi0Q_AUoAXoECAIQAw&biw=1680&bih=854&dpr=2#imgrc=FA4K7zfJ5cTN7M

supunTE replied:
Change function name and file name 62831d4e600c6c062635c4cb3badab2db5ce7f61

pasinducw commented on libs/util/src/lib/calculate-timeout-timer.ts:
```diff
@@ -0,0 +1,20 @@
+const averageWPM = AVERAGE_WPM;
+
+/** Calculating timeout timer for the race text.
+ *
```
Instead of the comments you've added inside, you can do the following in the function description:
==
Computes the time it would take for an average person to type the given block of text, and
returns it as the maximum allowed duration.

pasinducw replied:
Now, when you write up this description, you'll start noticing what's wrong in the approach that has been used to compute the time out duration

supunTE replied:
updated the function 9b960edbf7f516c3679f74a1f1fb1aae9a717cb4

pasinducw commented on libs/util/src/lib/calculate-timeout-timer.spec.ts:
```diff
@@ -0,0 +1,11 @@
+describe('[Utils] calculateTimeoutTimer', () => {
+  it.each([[M_RACE_TEXT0], [M_RACE_TEXT1]])('Calculate timeout timer', text => {
+    const timeoutDuration = calculateTimeoutTimer(text);
+    expect(timeoutDuration).toBeGreaterThanOrEqual(60);
```
why the range instead of absolute times?

supunTE replied:
changed
b3140266ae73aee73a842b6ebf3144f18e0b9239

pasinducw commented on libs/util/src/lib/extract-ids.spec.ts:
```diff
@@ -0,0 +1,92 @@
+        try {
+          extractId(inputId, inputIdType, outputIdType);
+        } catch (e) {
+          expect((e as Error).message).toBe('Invalid type');
```
This is not how you evaluate if a function throws. Notice that if the function did not throw, you are never going to reach line 87, and nothing will be evaluated.

See https://jestjs.io/docs/expect#tothrowerror

supunTE replied:
Solved d4b25066ee5b72d3895f2a05260396ef1b1ea054

pasinducw commented on libs/util/src/lib/extract-ids.ts:
```diff
@@ -0,0 +1,101 @@
+} from '@razor/models';
+
+export enum ExtractIdType {
+  tournament = 'tournament',
```
Please use PascalCase for the enum keys

supunTE replied:
Changed. f527eed17585e459a737ae2a1503f6d4adae8c6f

pasinducw commented on libs/util/src/lib/extract-ids.ts:
```diff
@@ -0,0 +1,101 @@
+      } else {
+        throw new Error('Invalid type');
+      }
+    // Output id type is "player".
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/extract-ids.ts:
```diff
@@ -0,0 +1,101 @@
+  const splitedId = inputId.split('-');
+  // Switching by output id type.
+  switch (outputIdType) {
+    // Output id type is "tournament".
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/extract-ids.ts:
```diff
@@ -0,0 +1,101 @@
+
+  // Split id into parts by '-'.
+  const splitedId = inputId.split('-');
+  // Switching by output id type.
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/extract-ids.ts:
```diff
@@ -0,0 +1,101 @@
+    throw new Error('Invalid input value');
+  }
+
+  // Split id into parts by '-'.
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/extract-ids.ts:
```diff
@@ -0,0 +1,101 @@
+
+  // Check the validity of the input id.
+  const validInput = checkValidityOfId(inputIdType, inputId);
+  // If invalid
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/extract-ids.ts:
```diff
@@ -0,0 +1,101 @@
+      } else {
+        throw new Error('Invalid type');
+      }
+    // Output id type is "race"
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.spec.ts:
```diff
@@ -0,0 +1,63 @@
+      raceTextLength,
+    );
+    expect(leaderboard).toEqual(
+      expect.arrayContaining([
```
If the array had additional elements, beyond what you have specified here, the test will not fail, right?

So, you've verified that the expected users are getting included, but missed to verify whether non-expected users are excluded.

supunTE replied:
Update test to check on complete leaderboard
5aa1fee51a7c524cb4e36d38e1681d4b8aa05db1

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+  /** Entries for incomplete players of race. */
+  const timeoutEntries: AppTimeoutLeaderboardEntry[] = [];
+
+  /** For every player-log-id of player logs */
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+  /** For every player-log-id of player logs */
+  let playerLogId: AppPlayerLogId;
+  for (playerLogId in playerLogs) {
+    /** Race id of relevant player log id. */
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+      ExtractIdType.race,
+    );
+
+    /** Player id of relevant player log id. */
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+
+    // Check whether the race owns the race log.
+    if (raceIdOfPlayerLog === raceId) {
+      /** Length of the current player log. */
```
Unnecessary. The variable name describes this comment

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+    if (raceIdOfPlayerLog === raceId) {
+      /** Length of the current player log. */
+      const playerLogsLength = playerLogs[playerLogId].length;
+      /** Last logged text length of the player. */
```
Unnecessary. Variable name already conveys the meaning

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+ * WPM is only calculated for the race-completed players.
+ * Term Quaater is considered between two markers of race text length.
+ *
+ * @param {number} length - Length of the text.
```
Data type is unnecessary. We already do describe the type with typescript.

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+ * Term Quaater is considered between two markers of race text length.
+ *
+ * @param {number} length - Length of the text.
+ * @param {number} timestamp2 - Player logs.
```
Data type is unnecessary.
What is timestamp2?

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+ * @returns {number} - Average wpm.
+ */
+const calculateWPM = (length: number, logs: AppPlayerLog[]): number => {
+  /** First marker(checkpoint) for the race text length */
```
Instead of having 3 lines of comments for each of the 3 marks. You can write a single line, so the content will be concise.

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+   */
+  const textLengthCheckpoints = [0, 0, 0, 0, length];
+
+  // For every player log
```
Unnecessary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+  //  textLengthCheckpoints = [ 0, 120, 245, 375, 500 ]
+  //
+  for (let index = 0; index < logs.length; index++) {
+    // Assigning player timestamp marker(checkpoint) & player text length marker(checkpoint)
```
Instead of repeating the same comment 4 times, add a single line summary

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on libs/util/src/lib/generate-leaderboard.ts:
```diff
@@ -0,0 +1,224 @@
+
+/** Calculate WPM for a quarter
+ *
+ * @param {number} timestamp1 - First timestamp.
```
Update the variable names. Drop the types. 
Also - if what you can describe with the comment is just the same as code, drop that comment; it's probably unnecessary.

supunTE replied:
Fixed b5eb08613ccaa263897da9c31b3c47b8df9886ee

pasinducw commented on nx.json:
```diff
@@ -9,7 +9,7 @@
       "options": {
         "cacheableOperations": ["build", "lint", "test", "e2e"],
-        "accessToken": "NjQ4ZDk0NjItY2E2Ni00ODE2LWE4ZjctNzlmNGVkNmQ0MTIwfHJlYWQtd3JpdGU="
+        "accessToken": "NGEzYTI0NmUtN2IxNS00OTIwLTgwMmYtYzM5YWY4NWRmMzJhfHJlYWQtd3JpdGU="
```
What's up with this token? Why is it tracked in git?

supunTE replied:
This token will keep nx logs connected to an account.
![image](https://user-images.githubusercontent.com/33794760/208398603-3f854eab-98f2-4fdc-8e77-d8889e078700.png)

I'll create a new access token and move it to the env

supunTE replied:
Done. The access token is removed from the nx.json file. I made an env file and add the token there to access it locally. 
 (**Nx cloud will be used only in local runs from now on.**)

supunTE replied:
62fb58b1fac17f0f9d575e958ac6a61d55449834
5e720001cf4bea424b0fac726d6ecff85efd442a

supunTE committed:
Merge branch 'the-ai-team:main' into feature/7-write-unit-tests

supunTE committed:
fix(util): remove unnecessary variable

supunTE committed:
docs(data models): update sample ids in docs

alphabet used to generate uid was changed.

supunTE committed:
refactor(effects): use constant to give zero padding to index of race

supunTE committed:
docs(constants): add docs to constants

supunTE committed:
feat(util): update calculate timeout function

Add extra time to the avaergae time can be taken. Update js docs

supunTE committed:
refactor(util): change calculate timeout timer function to compute race duration

supunTE committed:
style(husky): add new line at end on pre-push hook

pasinducw commented on libs/models/src/index.ts:
```diff
@@ -5,3 +5,8 @@ export * from './lib/state/leaderboard';
 export * from './lib/state/state-model';
 export * from './lib/state/log-message';
 export * from './lib/state/uid';
```
You can have an `index.ts` file within `lib/state` that exports the state models, and a similar one in `lib/sockets`.
Then this file would just have 2 exports (`./lib/state`, and `./lib/sockets`)

supunTE replied:
Solved. dc11cf4938f3e2d4a461fada7a807276d7a0a160

pasinducw commented on libs/util/src/lib/generate-avatar-link.ts:
```diff
@@ -1,16 +1,23 @@
-  playerName: string,
-): Promise<string> => {
+import { TextEncoder } from 'util';
+global.TextEncoder = TextEncoder;
```
What does this line do?

supunTE replied:
Removed. Global text encoder initialization was added to prevent an error while serving the react app. But the error cannot be reproduced anymore.

supunTE replied:
1e9960e3f5635b25908d286bf437999de16ad720

supunTE committed:
test(util): replace try catch with toThrow method to avoid skipping test without an error

re #7

supunTE committed:
chore(gitignore): add vscode settings json into gitignore

supunTE committed:
refactor(data models): export sockets and state models separately and then combine

make two separate index files for the sockets and state models and then combine them in the main index file

supunTE committed:
style(util): use PascalCase for enum keys in extract id type

supunTE committed:
docs(util): remove unnecessary lines

supunTE committed:
refactor(util): fix spelling mistakes

supunTE committed:
chore(nx cloud): move nx cloud access token to env

re #7

supunTE committed:
build(env): implement dotenv to transfer secret token while running scripts

supunTE committed:
test(util/generate-leaderboard): replace array containing check with a complete equal check

update test to check complete leaderboard without checking only contain items

re #7

supunTE committed:
test(util): replace check for absolute time in compute race duration

replace check for absolute time which was only check the range.

supunTE committed:
docs: clean jsdocs

remove unnecessary type declaring in jsdocs. type is already declared using ts.
